### PR TITLE
Colin items

### PR DESCRIPTION
**Pull request by colintalex**
-Creates test for item index to be visible to all users on system
-Adds links to images for items on item index page
-Creates most/least popular methods for Item model
-Creates section in view for most/least popular items
-Creates tests for most/least popular items

### CODE REVIEW

File: app/models/item.rb [hunk 1]
```diff
@@ -25,4 +25,11 @@ def no_orders?
     item_orders.empty?
   end
 
+  def self.most_popular
+    select('items.*, SUM(quantity) as total_quantity').joins(:item_orders).group('items.id').order('total_quantity DESC').limit(5)
+  end
+
+  def self.least_popular
+    select('items.*, SUM(quantity) as total_quantity').joins(:item_orders).group('items.id').order('total_quantity').limit(5)
+  end
```
Comment (DavidTTran) [inline app/models/item.rb:34]: It probably is a little redundant to have identical methods only for `DESC` to be changed. Helper method?

File: spec/features/items/index_spec.rb [hunk 2]
```diff
@@ -46,16 +53,121 @@
...
+      item_order1 = ItemOrder.create(order_id: order1.id, item: pull_toy, quantity: 1, price: pull_toy.price)
+      item_order2 = ItemOrder.create(order_id: order1.id, item: dog_bone, quantity: 2, price: dog_bone.price)
+      item_order3 = ItemOrder.create(order_id: order1.id, item: tire, quantity: 3, price: tire.price)
+      item_order4 = ItemOrder.create(order_id: order1.id, item: rim, quantity: 4, price: rim.price)
+      item_order5 = ItemOrder.create(order_id: order1.id, item: chew_rope, quantity: 5, price: chew_rope.price)
+      item_order6 = ItemOrder.create(order_id: order1.id, item: squeaky_toy, quantity: 6, price: squeaky_toy.price)
+      item_order7 = ItemOrder.create(order_id: order1.id, item: kong_toy, quantity: 7, price: kong_toy.price)
+      item_order8 = ItemOrder.create(order_id: order1.id, item: tire_parts, quantity: 8, price: tire_parts.price)
+      item_order9 = ItemOrder.create(order_id: order1.id, item: axle, quantity: 9, price: axle.price)
+      item_order10 = ItemOrder.create(order_id: order1.id, item: seat, quantity: 10, price: seat.price)
```
Comment (DavidTTran) [inline spec/features/items/index_spec.rb:146]: Yikes

File: spec/features/items/index_spec.rb [hunk 3]
```diff
@@ -46,16 +53,121 @@
...
-        expect(page).to have_content(@dog_bone.description)
-        expect(page).to have_content("Price: $#{@dog_bone.price}")
-        expect(page).to have_content("Inactive")
-        expect(page).to have_content("Inventory: #{@dog_bone.inventory}")
-        expect(page).to have_link(@brian.name)
-        expect(page).to have_css("img[src*='#{@dog_bone.image}']")
+    it "As any kind of user on the system, I can see all items except disabled items" do
+      user = User.create(name: "David", address: "123 Test St", city: "Denver", state: "CO", zip: "80204", email: "123@example.com", password: "password", role: 1)
+      merchant = User.create(name: "Colin", address: "123 Test St", city: "Denver", state: "CO", zip: "80204", email: "456@example.com", password: "password", role: 2)
+      admin = User.create(name: "Max", address: "123 Test St", city: "Denver", state: "CO", zip: "80204", email: "789@example.com", password: "password", role: 3)
```
Comment (DavidTTran) [inline spec/features/items/index_spec.rb:62]: I think it's implied that if a basic user doesn't see something than neither would merchant/visitor/admin but good testing

File: spec/features/items/index_spec.rb [hunk 4]
```diff
@@ -46,16 +53,121 @@
...
+      expect(page).to have_link(@pull_toy.name)
+
+      expect(page).to have_no_link(@dog_bone.name)
+
+      visit '/login'
+
+      fill_in :email, with: user.email
+      fill_in :password, with: user.password
+
+      click_on 'Login'
```
Comment (DavidTTran) [inline spec/features/items/index_spec.rb:76]: This feels a little redundant, maybe you can stub `:current_user` for these

File: db/seeds.rb [hunk 5]
```diff
@@ -16,13 +17,33 @@
...
+ItemOrder.create(order_id: order1.id, item: pull_toy, quantity: 1, price: pull_toy.price)
+ItemOrder.create(order_id: order1.id, item: dog_bone, quantity: 2, price: dog_bone.price)
+ItemOrder.create(order_id: order1.id, item: tire, quantity: 3, price: tire.price)
+ItemOrder.create(order_id: order1.id, item: rim, quantity: 4, price: rim.price)
+ItemOrder.create(order_id: order1.id, item: chew_rope, quantity: 5, price: chew_rope.price)
+ItemOrder.create(order_id: order1.id, item: squeaky_toy, quantity: 6, price: squeaky_toy.price)
+ItemOrder.create(order_id: order1.id, item: kong_toy, quantity: 7, price: kong_toy.price)
+ItemOrder.create(order_id: order1.id, item: tire_parts, quantity: 8, price: tire_parts.price)
+ItemOrder.create(order_id: order1.id, item: axle, quantity: 9, price: axle.price)
+ItemOrder.create(order_id: order1.id, item: seat, quantity: 10, price: seat.price)
```
Comment (Maxwell-Baird) [inline db/seeds.rb:49]: I wonder if there is an easier way to use the seeds in our test files

**Pull request merged**